apps/server/src/modules/authorization/authorization.service.ts

```diff
@@ -61,6 +63,27 @@ export class AuthorizationService extends BasePermissionManager {
 		}
 	}
 
+	async hasPermissionsByReferences(
```

Comment:
Why does this method exist? Couldn't we just use `hasPermissionByReference()`? Or even better just  load the entity before and use `hasPermission()`

Reply:
being able to check multiple permissions in a single call is for some usecases a desireable behavior. Having it encapsulated in the authorization service makes the most sense to us, as it's the services domain.

Entities shouldn't be loaded in usecases, only by repos (which is a big TODO currently). So loading those is seems counterproductive for that architectual goal.

Reply:
What use cases would that be then? Currently I can see no example.

But you would still load a domain object (instead of the entity) in the use case, wouldn't you? And this DO would then act as  the authorization subject. Same principle.

Reply:
Usecases include things like:
- Differentiate permissions for video conferences
- Differentiate permissions for teams
- File Storage Permissions

All those cases are tied to multiple, finely defined permissions and having a simple way to get check them is desireable imo.

About loading DOs/Entities it is a valid point. Once we operate on DOs, calls by reference might be not needed anymore.
Currently though i would not call other services with Entites. References keep it cleaner for UC/Service communications, and is in general undesirable for a DDD context to use raw Database Objects. I would like to keep those methods until then